Curved volume: move methods into ddgclib/, add meshes & benchmarks

songyideng opened the pull request:
Benchmarks are method-agnostic again; base/cases handle only data loading, timing, and summaries.

Curved-volume implementation lives in ddgclib/curved_volume/ with a thin façade ddgclib/_curved_volume.py.

Method registered in ddgclib/_method_wrappers.py as "curved_volume".

Added MshCase that loads a .msh and sets analytics to np.nan.

Analytic shapes keep true formulas; unknown analytics use np.nan.

Rotation/translation drivers are thin wrappers under ddgclib/curved_volume/.

Commit:
Curved volume: move methods into ddgclib/, add meshes & benchmarks

Stefan-Endres commented:
Thank you for the pull request, I'll work through the code now.

Just one immediate request: please add *.pyc to your gitignore to avoid churning in the future, we can delete these files for now (in general try to avoid uploading binaries).

Stefan-Endres commented on benchmarks/_benchmark_cases.py:
```diff
@@ -1,20 +1,54 @@
-        self.r_minor = r_minor
-        self.n_u = n_u
-        self.n_v = n_v
+        self.R_major = float(R_major)
```
This is not needed for the Python versions we are using, if you want to do static type analysis we should instead follow [PEP 484](https://peps.python.org/pep-0484/) for this.

Stefan-Endres commented on benchmarks/_benchmark_cases.py:
```diff
@@ -94,21 +105,7 @@ def dA(self, uv_i, uv_j, uv_k, n_samples=3):
-        float
-            Integrated mean curvature over area.
-        """
+        """Integrate mean curvature times area over a triangle in (u,v)-space."""
```
Please keep the numpy style documentations in everywhere, I don't see why this needs to be modified.

Stefan-Endres commented on benchmarks/_benchmark_classes.py:
```diff
@@ -1,149 +1,216 @@
             self.curvature_i = Curvature_i(self.method.get("curvature_i_method", "default"))
-        except NotImplementedError as e:
-            logger.warning(f"curvature_i method not implemented: {e}")
+        except Exception as e:
```
We definitely don't want to use the `Exception` base class here, in case there are other valid errors that we don't want capture and should be debugged instead.

songyideng commented:
Hi Stefan,

thanks for the feedback. I’ll add *.pyc (and __pycache__/) to the .gitignore so these files won’t be tracked in the future.

Best,
Songyi

Stefan-Endres commented on benchmarks/_benchmark_classes.py:
```diff
@@ -1,149 +1,216 @@
-        List of benchmark classes to instantiate and run.
-    method : dict, optional
-        Dictionary specifying method names for each evaluator. Example:
+    Instantiate and run all provided benchmark cases with the specified method registry names.
```
Same as before, please keep numpy style docstrings from before.

Stefan-Endres commented on ddgclib/_method_wrappers.py:
```diff
@@ -1,20 +1,52 @@
+# -----------------------------------------------------------------------------
+# Optional toy/demo methods from benchmarks/ (used as defaults/fallbacks)
+# -----------------------------------------------------------------------------
+try:
```
Please don't put exception handling here, if the library is unable to import these then something went very wrong.

Stefan-Endres commented on ddgclib/_method_wrappers.py:
```diff
@@ -29,153 +61,142 @@
         self._func = _curvature_i_methods[method]
 
     def __call__(self, HC, complex_dtype="vv"):
         """
```
Please keep L44-54 in

Stefan-Endres commented on ddgclib/_method_wrappers.py:
```diff
@@ -29,153 +61,142 @@
-        -------
-        np.ndarray
-            Area contribution per vertex.
+        HC: (points, simplices) if 'vf', otherwise a hyperct.Complex-like structure.
```
Same as above, keep old docstring

Stefan-Endres commented on ddgclib/_method_wrappers.py:
```diff
@@ -29,153 +61,142 @@
-        -------
-        np.ndarray
-            Area per triangle face.
+        HC: (points, simplices) if 'vf', otherwise a hyperct.Complex-like structure.
```
Docstring

Stefan-Endres commented on ddgclib/_method_wrappers.py:
```diff
@@ -29,153 +61,142 @@
-        -------
-        float
-            Total surface area.
+        HC: (points, simplices); complex_dtype must be 'vf'.
```
Docstring

Stefan-Endres commented:
Everything else looks good to me. We can merge after rolling back the minor changes. In general it should work just using the old file since nothing here breaks the old code.

songyideng committed:
Add files via upload

songyideng committed:
Add files via upload

songyideng committed:
old cases and classes

songyideng committed:
add _volume_curved

songyideng committed:
Delete ddgclib/__pycache__ directory

songyideng committed:
Delete benchmarks/__pycache__ directory

Commit:
chore: ignore .pyc/__pycache__ and remove them from the index

Commit:
Add .pyc and __pycache__/ to .gitignore

Commit:
Remove tracked .pyc and __pycache__ (stop bytecode churn)

songyideng commented:
Hi @Stefan-Endres — quick update and change summary:

.gitignore: added *.pyc and __pycache__/; removed previously committed bytecode from the branch to avoid churn going forward.

ddgclib/_method_wrappers.py: only added support for _volume_curved; otherwise unchanged from original.

benchmarks/_benchmark_cases.py: only added MshCase to the original file.

benchmarks/_benchmark_classes.py: reverted back to the original version.

Stefan-Endres commented:
Great! This looks good to me.
